noWord/common/DefaultTransformation.py

```diff
@@ -0,0 +1,74 @@
+
+
+def merge( input, params, context):
+
+    targetResource = {}
+
+    if isinstance( input, list):
+        for resourceName in block['resources']:
+            data = context.getResource(context.resources, resourceName)
+            targetResource.update(deepcopy(data))
+
+    return targetResource
+
+def slice( input, params, context):
+
+    inputRes = context.getResource(context.resources, input)
+
+    outputList = inputRes[params['start']:params['end']]
+
+    return outputList
+
+
+def distinct( input, params, context):
+
+    inputRes = context.getResource(context.resources, input)
+
+    outputRes = []
+    outputResList = []
+
+    for item in inputRes :
+        if params['key'] in item :
+            token = item[params['key']]
+            if not token in outputResList :
+                outputResList.append(token)
+                entry = {}
+                entry[params['key']] = token
+                outputRes.append(entry)
+
+    return outputRes
+
+
+def autonumber( input, params, context):

```

Comment:
The transformation modifies the input resource as well. This may lead to undesired behaviours.

Reply:
DONE